fix: enabled scanpopup at startup doesn't work

shenlebantongying opened the pull request:
fix https://github.com/xiaoyifang/goldendict/issues/361

main change:

From this line that only change UI widget's state:
```
enableScanningAction->setChecked( true );
```


to this that activate the action which change Both UI widget's state and setup scanning :
```
enableScanningAction->trigger();
```

xiaoyifang commented:
next time ,try to make the code rearrange  into another PR .

xiaoyifang commented:
`enableScanningAction->trigger();`
should put below the scanpop initiliazation.

xiaoyifang commented:
I think this bug is not existed before ,when it was introduced.

shenlebantongying commented:
This bug was introduced in https://github.com/xiaoyifang/goldendict/commit/497ecf3719edc84b2aaf54347cfeb3f88e541849

The code before has a connection between `QCliboard::changed` and `Mainwindow::clipboard_changed` forever, and keep checking the status of `enableScanningAction`.

The code after changed to connect when the `enableScanningAction` is enabled and disconnect when it is turned off. The signal `QAction::toggle(bool)` contains the on or off info.

Main reason of the change: on macOS, the clipboard don't have an event emitted on system level saying "clipboard changed" (thus QClipboard::changed won't be emitted unless app is activated), thus we have to keep monitoring it, and resume/pause the monitor. Keep reading the clipboard consume a lot of resources, thus we cannot have a connection betwen a connection between `QCliboard::changed` and `Mainwindow::clipboard_changed` forever like the old code.

I think the new code is superior, because Scanpopup's 💡 icon switching and connect/disconnect is now coupled together. The old code has related logic scattered everywhere.
